### ISSUE TITLE
CI simulation in the pit

### ISSUE THREAD

**Issue by ErikNatanael** [KTH/ci-hackathon#29]
Build an interactive installation for collaborative poetry based on the concept of CI.

**Comment by bbaudry** [comment]
That's super cool!
How about these possible inspirations for the initial text
- related to the hackathon's scientific orientation [Mind and Matter: The Intersection of Poetry and Science](https://blogs.plos.org/thestudentblog/2016/01/15/scienceandpoetry/)
- in reference to Sweden, a poem by [Gustaf Fröding](https://sv.wikipedia.org/wiki/Gustaf_Fr%C3%B6ding)
- in reference to pop culture in R1, the lyrics of [Nothing really matters](http://www.metrolyrics.com/nothing-really-matters-lyrics-madonna.html)

**Comment by bbaudry** [comment]
[The poetry of science](https://thepoetryofscience.scienceblog.com/)

**Comment by tdurieux** [comment]
It could be really nice to send something to 'The poetry of science' at the end of the Hackaton!

**Comment by bbaudry** [comment]
indeed, [Sam Illingworth](https://www.samillingworth.com/) might be intrigued by the process of CI poetry

**Comment by ErikNatanael** [comment]
That's a very cool! If he accepts outside contributions that would be great

**Comment by nadiacw** [comment]
Here are some of my references for the poetry:
http://www.ericascourti.com/
http://www.joannemcneil.com/projects-2/
https://pipthornton.com/2016/06/12/poem-py-a-critique-of-linguistic-capitalism/
http://poeticcomputation.info/
http://code-poems.com/index.html
https://sfpc.io/
https://news.stanford.edu/news/2013/december/code-poetry-slam-122013.html (next hackathon idea - software poetry slam)
https://blog.massol.me/all-work-and-no-play-kpiv/

**Comment by bbaudry** [comment]
some code for the code poetry slam is on github
https://github.com/lwu/say-23
https://github.com/lwu/say-23/blob/master/say23.english

could fun to use it and to propose the version produced on Monday evening as a PR

**Comment by bbaudry** [comment]
@nadiacw and @ErikNatanael , the installation was a success!
Can you push the code and the poem on the repo?

**Comment by tdurieux** [comment]
We can do something like this: https://githistory.xyz/ but nicer to see the evolution of the text :)

**Comment by ErikNatanael** [comment]
Good idea! That's pretty cool, will certainly serve as inspiration for a visualisation :D

Thomas Durieux <notifications@github.com> skrev: (18 oktober 2019 15:01:29 CEST)
>We can do something like this: https://githistory.xyz/ but nicer to see
>the evolution of the text :)

-- 
https://eriknatanael.com/
Sent from a mobile device.

**Comment by ErikNatanael** [comment]
Let us know if you take a stab at it! :)

**Comment by tdurieux** [comment]
First version online: https://kth.github.io/ci-hackathon/solutions/tdurieux_poet_history/index.html

It is really simple for the moment. I am open to any suggestion to make it better :)

**Comment by bbaudry** [comment]
it's super cool!

**Comment by ErikNatanael** [comment]
That's brilliant! :D Only tiny suggestions and ideas:
- maybe "poem" in the title instead of "Poet"?
- slightly faster tempo could be cool maybe?
- Do you think it would be nice with the "standby" ambient version of the sonification in the background?
- Possibly a timeline where you could "scrub" along the versions would be cool? The minimal look is great so it would have to be a very very tasteful slider
- I played around with putting the title on the side, I think it looks kind of nice, but it's pretty mobile hostile so I don't know :)
![poem-history-fooling-around](https://user-images.githubusercontent.com/12977989/67240655-311d2500-f452-11e9-9f47-a603e6e7cbda.png)

**Comment by nadiacw** [comment]
Amazing!
I agree with Erik on changing the title to "poem". Also, the font used in the printed poems of the installation is https://fonts.google.com/specimen/Karma, so it would be super nice to use that.

**Comment by Jacarte** [comment]
Super cool Thomas !!

**Comment by tdurieux** [comment]
> maybe "poem" in the title instead of "Poet"?

Yes, changed

> slightly faster tempo could be cool maybe?

Done

> Do you think it would be nice with the "standby" ambient version of the sonification in the background?

It can be interesting, where can I find the music?

> Possibly a timeline where you could "scrub" along the versions would be cool? The minimal look is great so it would have to be a very very tasteful slider

I did a first version, WDYT?

> I played around with putting the title on the side, I think it looks kind of nice, but it's pretty mobile hostile so I don't know :)

I used your idea, I can easily adapt the team for mobile but not sure if there are a lot of mobile users that will see that page :)

**Comment by ErikNatanael** [comment]
Very nice! Yeah, I agree, let's leave mobile users to their fate! I'll make a recording of some background music.
I love the slider, except for the scrollbars. I think if both they and the poem were centered that would be neat (I hacked it for my resolution):
![Screenshot_20191022_170216](https://user-images.githubusercontent.com/12977989/67300438-39c33900-f4ee-11e9-897f-55c2909e2511.png)
(p.s. it definitely still says "poet" in the title for me)

**Comment by tdurieux** [comment]
I centered the content. Also, if you maintain a click on the poem the animation is paused. It is resumed when you release the click. It allows reading the poem.

**Comment by ErikNatanael** [comment]
It's beautiful! Can I include it in the archive page for the hackathon?

**Comment by tdurieux** [comment]
Of course, and if you want to change something please do it :)

**Comment by nadiacw** [comment]
Super cool! Thank you so much for doing this!
I just wonder if it would be nice to pause the scroll somehow, to be able to read each poem? The overall effect of constant change is one of the goals, but the installation was also designed so each instance could be read and make (maybe?) sense.
But it's great how it is!

**Comment by tdurieux** [comment]
You can pause the scroll my maintaining a click on the web page. I don't really know how show the evolution and at the same time highlight the poems.

Maybe randomly stop and start to read the poem?

**Comment by ErikNatanael** [comment]
One idea is to invert the colours (with a nice fade :D ) when it is in the pause mode just to highlight it to the viewer. Perhaps have one click outside of the timeline toggle between pause and play? I don't know

Thomas Durieux <notifications@github.com> skrev: (23 oktober 2019 13:53:03 CEST)
>You can pause the scroll my maintaining a click on the web page. I
>don't really know how show the evolution and at the same time highlight
>the poems.
>
>Maybe randomly stop and start to read the poem?

-- 
https://eriknatanael.com/
Sent from a mobile device.

**Comment by tdurieux** [comment]
Good idea to change the color. 
I added that and now the poem is read when you pause the "scroll".

**Comment by ErikNatanael** [comment]
Looks nice! Haha, is that a Norwegian TTS voice?

**Comment by tdurieux** [comment]
It uses the default language of your browser :) mine is in English

**Comment by tdurieux** [comment]
I changed this behavior, now it is forcing an English voice.

**Comment by gluckzhang** [comment]
Just tried the latest version. Super cool!

**Comment by tdurieux** [comment]
@ErikNatanael & @nadiacw I just realized that I forgot to cite you in the visualization. It is now fixed :)

**Comment by ErikNatanael** [comment]
Thanks, I definitely think you should add your own name too though!!
